pkg/cmd/triggerbinding/describe_test.go

```diff
@@ -0,0 +1,178 @@
+	tbs := []*v1alpha1.TriggerBinding{
+		tb.TriggerBinding("tb1", "ns"),
+	}
+	cs := test.SeedTestResources(t, triggertest.Resources{TriggerBindings: tbs, Namespaces: []*corev1.Namespace{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ns",
+		},
+	}}})
+	p := &test.Params{Triggers: cs.Triggers, Kube: cs.Kube}
+
+	triggerBinding := Command(p)
+	out, err := test.ExecuteCommand(triggerBinding, "desc", "tb1", "-n", "ns")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	golden.Assert(t, out, fmt.Sprintf("%s.golden", t.Name()))
+}
+
+func TestTriggerBindingDescribe_WithParams(t *testing.T) {
+	tbs := []*v1alpha1.TriggerBinding{
+		tb.TriggerBinding("tb1", "ns",
+			tb.TriggerBindingSpec(
+				tb.TriggerBindingParam("key", "value"))),
+	}
+	cs := test.SeedTestResources(t, triggertest.Resources{TriggerBindings: tbs, Namespaces: []*corev1.Namespace{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ns",
+		},
+	}}})
+	p := &test.Params{Triggers: cs.Triggers, Kube: cs.Kube}
+
+	triggerBinding := Command(p)
+	out, err := test.ExecuteCommand(triggerBinding, "desc", "tb1", "-n", "ns")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	golden.Assert(t, out, fmt.Sprintf("%s.golden", t.Name()))
+}
+
+func TestTriggerBindingDescribe_WithOutputName(t *testing.T) {
+	tbs := []*v1alpha1.TriggerBinding{
+		tb.TriggerBinding("tb1", "ns",
+			tb.TriggerBindingSpec(
+				tb.TriggerBindingParam("key", "value"))),
+	}
+	cs := test.SeedTestResources(t, triggertest.Resources{TriggerBindings: tbs, Namespaces: []*corev1.Namespace{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ns",
+		},
+	}}})
+	p := &test.Params{Triggers: cs.Triggers, Kube: cs.Kube}
+
+	triggerBinding := Command(p)
+	out, err := test.ExecuteCommand(triggerBinding, "desc", "-o", "name", "-n", "ns", "tb1")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	golden.Assert(t, out, fmt.Sprintf("%s.golden", t.Name()))
+}
+
+func TestTriggerBindingDescribe_WithOutputYaml(t *testing.T) {
+	tbs := []*v1alpha1.TriggerBinding{
+		tb.TriggerBinding("tb1", "ns",
+			tb.TriggerBindingSpec(
+				tb.TriggerBindingParam("key", "value"))),
+	}
+	cs := test.SeedTestResources(t, triggertest.Resources{TriggerBindings: tbs, Namespaces: []*corev1.Namespace{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ns",
+		},
+	}}})
+	p := &test.Params{Triggers: cs.Triggers, Kube: cs.Kube}
+
+	triggerBinding := Command(p)
+	out, err := test.ExecuteCommand(triggerBinding, "desc", "-o", "yaml", "-n", "ns", "tb1")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	golden.Assert(t, out, fmt.Sprintf("%s.golden", t.Name()))
+}
+
+func TestTriggerBindingDescribe_WithMultipleParams(t *testing.T) {
+	tbs := []*v1alpha1.TriggerBinding{
+		tb.TriggerBinding("tb1", "ns",
+			tb.TriggerBindingSpec(
+				tb.TriggerBindingParam("key1", "value1"),
+				tb.TriggerBindingParam("key2", "value2"),
+				tb.TriggerBindingParam("key3", "value3"),
+				tb.TriggerBindingParam("key4", "value4"))),
```

Comment:
Could you please make at least one or two of these params an Array type instead of String?

Reply:
There is no helper function implemented to call in order to add Array
So i have already added a PR tektoncd/triggers#543 to handle that
I will add testcase once that PR merge

Reply:
My apologies for the confusion. I noticed there is an Array type check in the template, but it seems that only strings are supported at this time: https://github.com/tektoncd/triggers/pull/543#issuecomment-616490309. I think we can remove the Array type from the template, and this pr should be good. Thanks for looking into this though and trying.

Reply:
No worries at all atleast it gets cleared